src/Recursor.h

```diff
@@ -611,17 +611,17 @@ void Recursor<Derived>::ExtendBeta(const M& beta, size_t lastColumn, M& ext, int
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn + 1);
     else
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn);
-    for (int j = 0; j <= lastColumn && j <= numExtColumns; ++j)
+    for (size_t j = 0; j <= lastColumn && j <= numExtColumns; ++j)
         beginRow = std::min(static_cast<int>(beta.UsedRowRange(lastColumn - j).first), beginRow);
 
-    for (int j = lastColumn; j + numExtColumns > lastColumn; j--) {
+    for (int j = lastColumn; j + numExtColumns - lastColumn > 0; j--) {
         /* Convert from old template to new template coordinates.
            lengthDiff will be 0 for substitution, -1 for deletion and +1 for
            insertion
          */
-        int jp = j + lengthDiff;
+        size_t jp = j + lengthDiff;
```

Comment:
okay here's a potential problem. lastColumn could theoretically be 0 along with lengthDiff of -1.

Okay, let's see if I can work this out:

j = lastColumn
numExtColumns = 1 + lengthDiff + lastColumn
j + numExtColumns - lastColumn > 0
j + numExtColumns - lastColumn = j + (1 + lengthDiff + lastColumn) - lastColumn = j + 1 + lengthDiff
j + 1 + lengthDiff > 0
j + lengthDiff > -1
j + lengthDiff >= 0

Okay it should be fine. We guard against it.